app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new params[:post]
+  end
+
+  def edit
+    @post = Post.find_by id: params[:id]
+  end
+
+  def update
+    @post = Post.find_by id: params[:id]
+
+    if @post.update post_params
+      flash[:success] = t "post.update"
+      redirect_to @post
+    else
+      flash.now[:warning] = t "post.updatefail"
+      render :edit
+    end
+  end
+
+  def show
+    @post = Post.find_by id: params[:id]
+  end
+
+  def create
+    @post = Post.new post_params
+
+    if @post.save
+      flash[:success] = t "post.create"
+      redirect_to @post
+    else
+      flash.now[:warning] = t "post.createfail"
+      render :new
+    end
+  end
+
+  def destroy
+    @post = Post.find_by id: params[:id]
```

Comment:
`@post = Post.find_by id: params[:id]` is wrote too many times, you should write in a private method and use before_action for actions that use @post